Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -102,30 +102,57 @@ open class ChatChannelListItemView: _View, ThemeProvider, SwiftUIRepresentable {
         guard let content = content else { return nil }
         if let typingUsersInfo = typingUserString {
             return typingUsersInfo
-        } else if let latestMessage = content.channel.latestMessages.first {
-            return "\(latestMessage.author.name ?? latestMessage.author.id): \(latestMessage.textContent ?? latestMessage.text)"
+        } else if let previewMessage = content.channel.previewMessage {
+            guard previewMessage.type != .system else {
+                return previewMessage.text
+            }
+            
+            let authorName = previewMessage.isSentByCurrentUser
+                ? L10n.you
+                : previewMessage.author.name ?? previewMessage.author.id
+            
+            let text = previewMessage.textContent ?? previewMessage.text
+            
+            return "\(authorName): \(text)"
         } else {
             return L10n.Channel.Item.emptyMessages
         }
     }
 
     /// Text of `timestampLabel` which contains the time of the last sent message.
     open var timestampText: String? {
-        if let lastMessageAt = content?.channel.lastMessageAt {
-            return timestampFormatter.format(lastMessageAt)
+        if let timestamp = content?.channel.previewMessage?.createdAt {
+            return timestampFormatter.format(timestamp)
         } else {
             return nil
         }
     }
+    
+    /// The delivery status to be shown for the channel's preview message.
+    open var previewMessageDeliveryStatus: ChatMessageDeliveryStatusCheckmarkView.Content? {
```

Comment:
Doesn't make sense to couple this view with `ChatMessageDeliveryStatusCheckmarkView`. Also, the name of this property doesn't reflect what actually returns. I think is better to return just the `MessageDeliveryStatus`
```suggestion
    open var previewMessageDeliveryStatus: MessageDeliveryStatus? {
```

Reply:
Creating `ChatMessageDeliveryStatusCheckmarkView.Content` is just a initializer. One thing is to create this data structure, another is the logic when we should create it. We shouldn't couple both. Right now `previewMessageDeliveryStatus` is responsible for 2 things, deciding whether it should display the delivery status, and creating the content of the `ChatMessageDeliveryStatusCheckmarkView`. IMO it should only decide whether it should display the delivery status.

I'm not saying to calculate the logic 3 times. This what I mean:

```swift
    /// The delivery status to be shown for the channel's preview message.
    open var previewMessageDeliveryStatus: MessageDeliveryStatus? {
        guard
            let content = content,
            let deliveryStatus = content.channel.previewMessage?.deliveryStatus
        else { return nil }
        
        switch deliveryStatus {
        case .pending, .failed:
            return deliveryStatus
        case .sent, .read:
            guard content.channel.config.readEventsEnabled else { return nil }
            
            return deliveryStatus
        default:
            return nil
        }
    }
```

Then the creation of the content is in the `updateContent()` function, there's no need to create the content in the property above. If a customer wants to reuse this logic but doesn't want to use `ChatMessageDeliveryStatusCheckmarkView.Content`, it won't be possible with the current solution.

Reply:
Done